### PR TITLE
dep-check: change to use sync methods

### PR DESCRIPTION
**Pull request by powerivq**
Did not notice comments on https://github.com/ampproject/amphtml/pull/35367

This PR addresses it.

### CODE REVIEW

File: build-system/tasks/dep-check.js [hunk 1]
```diff
@@ -173,12 +173,12 @@ const rules = depCheckConfig.rules.map((config) => new Rule(config));
  */
 async function getEntryPointModule() {
   const coreBinaries = ['src/amp.js', '3p/integration.js'];
-  const extensions = await fs.promises.readdir('extensions');
+  const extensions = fs.promises.readdirSync('extensions');
```
Comment (danielrozenberg) [inline build-system/tasks/dep-check.js:176]:
```suggestion
  const extensions = fs.readdirSync('extensions');
```

File: build-system/tasks/dep-check.js [hunk 2]
```diff
@@ -173,12 +173,12 @@ const rules = depCheckConfig.rules.map((config) => new Rule(config));
...
   const extensionEntryPoints = extensions
     .map((x) => `extensions/${x}`)
     .filter((x) => fs.statSync(x).isDirectory())
     .map(getEntryPoint);
-  const vendors = await fs.promises.readdir('3p/vendors');
+  const vendors = fs.promises.readdirSync('3p/vendors');
```
Comment (danielrozenberg) [inline build-system/tasks/dep-check.js:181]:
```suggestion
  const vendors = fs.readdirSync('3p/vendors');
```

**Review by danielrozenberg — CHANGES_REQUESTED** [review]

**Review by samouri — CHANGES_REQUESTED** [review]
I actually think this is less desirable than the async version (https://github.com/ampproject/amphtml/pull/35367#discussion_r692203170)

> My impression was that the await version was always preferred otherwise you block any other action that could happen concurrently. For example processing multiple getEntryPointModules concurrently would be sped up by using the async vs sync call

**Comment by powerivq** [comment]
@samouri Make sense. Closing this PR.

**Pull request closed without merging**